src/core_plugins/kibana/public/home/components/recently_accessed.js

```diff
@@ -0,0 +1,188 @@
+import './recently_accessed.less';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiPanel,
+  EuiLink,
+  EuiText,
+  EuiTextColor,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiIcon,
+} from '@elastic/eui';
+
+export const NUM_LONG_LINKS = 5;
+
+export class RecentlyAccessed extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false,
+    };
+  }
+
+  onButtonClick() {
+    this.setState({
+      isPopoverOpen: !this.state.isPopoverOpen,
+    });
+  }
+
+  closePopover() {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  }
+
+  renderDropdown = () => {
+    const dropdownLinks = [];
+    for (let i = NUM_LONG_LINKS; i < this.props.recentlyAccessed.length; i++) {
+      dropdownLinks.push(
+        (
+          <li
+            style={{ marginBottom: 8 }}
+            key={this.props.recentlyAccessed[i].link}
+          >
+            <EuiLink
+              className="dropwdownLink"
+              href={this.props.recentlyAccessed[i].link}
+            >
+              {this.props.recentlyAccessed[i].label}
+            </EuiLink>
+          </li>
+        )
+      );
+    }
+
+    const openPopoverComponent = (
+      <EuiLink onClick={this.onButtonClick.bind(this)}>
+        <EuiTextColor color="subdued">
+          {`+${dropdownLinks.length}`}
+        </EuiTextColor>
+        <EuiIcon
+          type="arrowDown"
+          color="subdued"
+        />
+      </EuiLink>
+    );
+
+    return (
+      <EuiFlexItem
+        key="dropdown"
+        grow={false}
+      >
+        <EuiPopover
+          id="popover"
+          ownFocus
+          button={openPopoverComponent}
+          isOpen={this.state.isPopoverOpen}
+          closePopover={this.closePopover.bind(this)}
+          anchorPosition="downRight"
+        >
+          <ul>
+            {dropdownLinks}
+          </ul>
+        </EuiPopover>
+      </EuiFlexItem>
+    );
+  }
+
+  renderLongLink = (link, label, includeSeperator = false) => {
+    let seperator;
+    if (includeSeperator) {
+      seperator = (
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type="dot"
+            color="subdued"
+          />
+        </EuiFlexItem>
+      );
+    }
+    // Want to avoid a bunch of white space around items with short labels (happens when min width is too large).
+    // Also want to avoid truncating really short names (happens when there is no min width)
+    // Dynamically setting the min width based on label lengh meets both of these goals.
+    const EM_RATIO = 0.6; // 'em' ratio that avoids too much horizontal white space and too much truncation
+    const minWidth = (label.length < 8 ? label.length : 8) * EM_RATIO;
+    const style = {
```

Comment:
Can we create a custom class for these styles instead of using an inline style? This will make it a little easier for engineers and designers to find these styles since they'll live in an expected location (e.g. LESS file or SCSS file once we support it).

Can we also add a `max-width: 300px` or something around that size? That will help avoid these kinds of unbalanced layouts:

![image](https://user-images.githubusercontent.com/1238659/36004707-e39674dc-0ce8-11e8-96b8-5672f3d81c72.png)

Reply:
Also, maybe we should add a tooltip to show the full name of each item when you hover?

Reply:
How would you dynamically set a min-width via less? A variable min-width really helps avoid lots of extra white space around really short names.

Reply:
Ah, shoot, I was skimming and overlooked that requirement. Sorry. Yeah that's a tough one... I think your solution is probably the best way to go about it.

Reply:
@cjcenizal As for the tooltip idea, anything that could ever get truncated needs to have it's text in a title attribute. That way, on hover you will always get the full text.